### ISSUE TITLE
Don't remove spaces in `calc`

### ISSUE THREAD

**Issue by nic-hartley** [kaj/rsass#53]
Repro, assuming Bash and the `rsass` binary somewhere on your `$PATH`:

```bash
echo $'main {\n  width: calc(100% - 20px);\n}' > test.sass
rsass test.sass
```

Importantly, the input is `calc(100% - 20px)` (with spaces); the output is `calc(100%-20px)` (without them) -- this makes the CSS invalid. 

#42 is possibly related, since this bug doesn't occur with non-function-like stuff (e.g. `border: 1px solid black` has all the spaces you'd expect) and special-casing `calc` to be excluded from function interpretation might fix this. I don't know enough about the code to be confident one way or the other.

Incidentally, there _is_ a workaround: Wrap the whole expression in an interpolation, e.g. `#{"calc(100% - 20px)"}` -- this preserves the spaces.

**Comment by kaj** [comment]
Yes, this is a problem.  I see two potential ways to fix it:

1. Recognize that the `calc` function is special (as specified in #42).  The arguments to `calc` should probably be kept verbatim except for interpolations.
2. Add general support for keeping whitespace around the `+` and `-` operators.  This already exists for the `*` and `/` operators.

I guess (1) is probably the most correct, as e.g. `calc(2px + 2px + 5%)` should probably be kept as is, and not evaluated to `calc(4px + 5%)` as it would if we go the (2) route.  Also, there are other reasons to fix #42, so doing that would give a "free" fix for this. On the other hand, (2) might be an easier fix for this specific issue.

**Comment by kaj** [comment]
No difference that I'm aware of, but the [sass spec](https://sass-lang.com/documentation/syntax/special-functions#calc-element-progid-and-expression) says:
> Sass allows any text in these function calls, including nested parentheses. Nothing is interpreted as a SassScript expression, with the exception that interpolation can be used to inject dynamic values.
